lib/backup/users.rb

```diff
@@ -0,0 +1,18 @@
+module Backup
+  class Users
+    attr_reader :user_dump_file
+
+    def initialize
+      @user_dump_file = File.join(Gitlab.config.backup.path, 'users.csv')
+    end
+
+    def dump
+      CSV.open(user_dump_file, "w") do |csv|
+        csv << ["email", "name", "admin", "projects_limit", "username", "can_create_group", "state"]
+        User.all.each do |u|
```

Comment:
For more efficient memory usage, use `User.find_each` instead of `User.all.each`